### PR TITLE
suppress __init__ deprecation warning

### PR DESCRIPTION
**Pull request by naxingyu**
build and test pass.

### CODE REVIEW

**Comment by csukuangfj** [comment]
+2
I didn't receive the `FutureWarning` messages because I added `-DNDEBUG` and removed `-g` from `kaldi.mk` while building kaldi.

I rebuilt kaldi yesterday and got the warnings; I think you got the solution from
https://github.com/pybind/pybind11/blob/master/docs/upgrade.rst#new-api-for-defining-custom-constructors-and-pickling-functions

**Comment by naxingyu** [comment]
Yes, that warnings only appear in debug mode.

**Comment by csukuangfj** [comment]
I used
```python
import warnings
warnings.simplefilter(action="ignore", category=FutureWarning)
```
to suppress it explicitly. Your solution is more elegant.

File: src/pybind/tests/test_kaldi_pybind.py [hunk 1]
```diff
@@ -28,9 +28,9 @@ def test_float_vector(self):
         gold = np.array([0, 0, 2, 2, 2])
         self.assertTrue((np_array==gold).all())
 
-    def test_float_matrxi(self):
+    def test_float_matrix(self):
         # test FloatMatrix
-        print("=====Testing FloatVector in kaldi_pybind=====")
+        print("=====Testing FloatMatrix in kaldi_pybind=====")
```
Comment (csukuangfj) [inline src/pybind/tests/test_kaldi_pybind.py:33]: this file is not properly formatted.
For instance, the following statement

```python
kp_matrix = kp.FloatMatrix(4,5)
```

There should be a space after the comma, i.e., `4, 5`.

After fixing the typos, could you also fix the format ?

Comment (naxingyu) [reply src/pybind/tests/test_kaldi_pybind.py:33]: sure. done.

**Pull request merged**